treecorr/util.py

```diff
@@ -234,6 +260,9 @@ def gen_multi_write(file_name, col_names, group_names, columns,
                 s = '## %s: %d\n'%(name, len(cols[0]))
                 fid.write(s.encode())
                 gen_write_ascii(fid, col_names, cols, params, precision=precision)
+    elif file_type == "HDF":
+        raise NotImplementedError("Cannot currently write 3-point correlations or "
+                                  "other multi-part data to HDF files")
```

Comment:
Since you bothered to implement an optional groupname parameter for `get_write_hdf`, this is straightforward, so it seems like we might as well implement it:
```
with h5py.File(file_name, 'w') as hdf:
    for name, cols in zip(group_names, columns):
        gen_write_hdf(hdf, col_names, cols, params, groupname=name)
```
It would only be relevant for output files of the new 3pt CrossCorrelation classes.  But if we're going to allow hdf files for output files, then we should probably add it for both varieties.  I won't make you wade into writing a unit test for it though.  I can add this after merging.

Reply:
Cheers - didn't realise this was that easy. I've added.